__tests__/endpoints.test.js

```diff
@@ -1,150 +1,167 @@
+    const inc_votes = {
+      votes: 2,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(inc_votes)
+      .expect(200)
+      .then(({ body: { article } }) => {
+        expect(article).toEqual({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          author: "butter_bridge",
+          created_at: "2020-07-09T20:11:00.000Z",
+          topic: "mitch",
+          votes: 102,
+          body: "I find this existence challenging",
+        });
+      });
+  });
+
+  test("Status 200 - should return the updated object with decreased votes when passed a negative number", () => {
+    const inc_votes = {
+      votes: -50,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(inc_votes)
+      .expect(200)
+      .then(({ body: { article } }) => {
+        expect(article).toEqual({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          author: "butter_bridge",
+          created_at: "2020-07-09T20:11:00.000Z",
+          topic: "mitch",
+          votes: 50,
+          body: "I find this existence challenging",
+        });
+      });
+  });
+
+  test("Status 400 - should return an error if the request body votes missing", () => {
+    const inc_votes = {};
+    return request(app)
+      .patch("/api/articles/beans")
+      .send(inc_votes)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+
+  test("Status 400 - should return an error if the request body votes are not an integer", () => {
+    const inc_votes = {
+      votes: "beans",
+    };
+    return request(app)
+      .patch("/api/articles/beans")
+      .send(inc_votes)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+
+  test("Status 404 - should return an error message if article doesn't exist", () => {
+    return request(app)
+      .patch("/api/articles/999")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Not found");
+      });
+  });
+
+  test("Status 400 - should return an error message if the wrong data type is passed to the url", () => {
+    return request(app)
+      .get("/api/articles/beans")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+});
+
+describe("GET /api/users", () => {
+  test("Status 200 - should return with an array of objects containing a username", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body: { usernames } }) => {
+        expect(usernames).toBeInstanceOf(Array);
+        expect(usernames).toHaveLength(4);
+        usernames.forEach((user) => {
+          expect(user).toMatchObject({
+            username: expect.any(String),

```

Comment:
There are more keys than this on a user, please make sure you are testing that the objects are returning as they should.

Reply:
Done :)